lib/livebook_web/plugs/auth_plug.ex

```diff
@@ -51,6 +55,34 @@ defmodule LivebookWeb.AuthPlug do
     end
   end
 
+  @doc """
+  Checks if given connection or session is authorized.
+  """
+  @spec authorized?(Plug.Conn.t()) :: boolean()
+  def authorized?(%Plug.Conn{} = conn) do
+    # User has access only to specific app pages, so they do not have
+    # access to any pages guarded by this plug.
+
+    LivebookWeb.UserPlug.build_current_user(
+      get_session(conn),
+      conn.assigns.identity_data,
+      conn.assigns.user_data
+    ).restricted_apps_groups == nil
```

Comment:
We are missing `authenticated?` here as well. (a updated the comment, since the condition is different than before):

```suggestion
    # Note that if the user has access restricted to specific app pages,
    # they are not authorized and have no access to any pages guarded
    # by this plug.

    authenticated?(conn) and
      LivebookWeb.UserPlug.build_current_user(
        get_session(conn),
        conn.assigns.identity_data,
        conn.assigns.user_data
      ).restricted_apps_groups == nil
```